### ISSUE TITLE
Not copying binary frameworks, OmniGraffle typeIdentifier issue

### ISSUE THREAD

**Issue by chriscomeau** [Carthage/Carthage#3093]
* carthage install method: [] .pkg, [] homebrew, [x] source
* `which carthage`: /usr/local/bin/carthage
* `carthage version`: 0.36.0
* `xcodebuild -version`: Xcode 12.2 Build version 12B45b
* Are you using `--no-build`?  No
* Are you using `--no-use-binaries`?  No
* Are you using `--use-submodules`?  No
* Are you using `--cache-builds`?  No
* Are you using `--new-resolver`?  No

**Cartfile**
```
#simple binary test
binary "https://dl.google.com/geosdk/GoogleMaps.json" == 4.0.0
```

**Carthage Output**
Test command `carthage boostrap`
```
*** xcodebuild output can be found in /var/folders/d5/dpdtwfrx2hzbzgcdytpr30800000gq/T/carthage-xcodebuild.Gqh7lk.log
*** Downloading binary-only framework GoogleMaps at "https://dl.google.com/geosdk/GoogleMaps.json"
```

**Actual outcome**
Carthage did not copy frameworks to `Build` folder.

**Expected outcome**
Carthage should copy frameworks to `Build` folder.

Currently running macOS 11.1 and Xcode 12.2.

**Issue**

After multiple tests and debugging, I figured out that it's an issue with my setup, but not sure how to fix it, or if it's worth adding a workaround to Carthage in case it happens other people.

In `frameworksInDirectory`, `filesInDirectory` is called with `kUTTypeFramework` (`com.apple.framework`) as the `typeIdentifier`, to filter only frameworks.

For some reason on my computer, when I debug and get to this point, the typeIdentifier of framework paths have a type of `com.omnigroup.foreign-types.apple-framework`. Probably related to having an old macOS sytem that i've been upgrading/restoring for years. 

**Tests**
* Tried deleting OmniGraffle
* Tried to set framework files in Get Info to be opened with Xcode, which it was already (and Change All...)
* Just using the terminal command `mdls` shows all framework files as `com.omnigroup.foreign-types.apple-framework` on my computer, but on a second computer correctly shows `com.apple.framework`.
* Tried to reset the system's filetypes (but not too familiar with the typeIdentifer system)
```
//https://superuser.com/questions/1401355/how-to-reset-the-icon-associated-to-a-given-file-type

/System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/Support/lsregister -kill -r -domain local -domain system -domain user
```
```
//https://apple.stackexchange.com/questions/329910/reset-uniform-type-identifiers-on-mac-os-x

/System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/LaunchServices.framework/Versions/A/Support/lsregister -kill -seed
```
Also tried with the [OnyX](https://www.titanium-software.fr/en/onyx.html) utility.

**Workaround**
A quick workaround/hack I did temporarily, in `filesInDirectory`, is to detect a failed check for `com.apple.framework` and to also check for `com.omnigroup.foreign-types.apple-framework`:
```
return url.typeIdentifier
                    .analysis(ifSuccess: { identifier in
                        //hack
                        var conforms = UTTypeConformsTo(identifier as CFString, typeIdentifier as CFString)
                        if conforms == false && typeIdentifier == "com.apple.framework" {
                            conforms = UTTypeConformsTo(identifier as CFString, "com.omnigroup.foreign-types.apple-framework" as CFString)
                        }
                        return conforms
                    }, ifFailure: { _ in
                        return false
                    })
```

**Question**
Any thoughts on how to fix the root problem of my machine, or would it be worth it to have a clean workaround in Carthage? Thanks!

**Comment by chriscomeau** [comment]
Running this command (and rebooting) seems to have fixed my issue, seeing `com.apple.framework` correctly now:

```
/System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/LaunchServices.framework/Versions/A/Support/lsregister -kill -r -domain u -domain s -domain l -v
```

Closing the issue.

**Comment by chriscomeau** [comment]
A few more details from Omni support:

> I see you managed to solve your trouble from your GitHub issue. I did want to share that this stemmed from an issue in OmniGraffle/a change in macOS Big Sur. We support importing .framework files and as such declare a filetype for them. This wasn't an issue on earlier versions of macOS, but on Big Sur it was leading to various issues like what you saw. We have since corrected this in OmniGraffle 7.18.1, but getting back out of the bad state can require extra effort as you saw even after removing the app.
> 
> WIth that, if you haven't already done so, you should be safe to reinstall OmniGraffle again.
